online-shopping-website/src/Components/ProductGrid.js

```diff
@@ -0,0 +1,54 @@
+import Card from '@mui/material/Card';
+import CardActions from '@mui/material/CardActions';
+import Button from '@mui/material/Button';
+import Grid from '@mui/material/Grid';
+
+const ProductPreview = (props) => {
+  return (
+    <Card className="ProductContainer">
+      <Grid container>
+        {/* First column of Product Card. Image renders on top of Card on lg and smaller screens. Renders on the left for lg and xl screens. */}
+        <Grid item xs={12} xl={6} sx={{ textAlign: 'center' }}>
+          <div className="ProductImageHelper"></div>
+          <img className="ProductImage" src={props.product.image} alt={props.product.name} />
+        </Grid>
+
+        {/* Second column. Contains all text and Buttons. */}
+        <Grid item xs={12} sm container>
+          <Grid item container direction="column">
+            <Grid item xs sx={{ paddingX: '10px' }}>
+              <h3>{props.product.name}</h3>
+              <p>{props.product.description}</p>
+              <h4>Sold by: {props.product.seller}</h4>
+              <h4>Price: {props.product.price} Ɖ</h4>
+              <CardActions sx={{ marginLeft: '-10px' }}>
+                <Button variant="contained" className="ProductButtonContained">Add to Cart</Button>
+                <Button variant="outlined" className="ProductButton">Learn More</Button>
+              </CardActions>
+            </Grid>
+          </Grid>
+        </Grid>
+      </Grid>
+    </Card>
+  );
+}
+
+// Creates a Grid item with a ProductPreview for each product in inputted data. Sets element keys to product ids.
+const iterateProduct = (data) => {
+  return data.products.map(product => {
+    return (
+      <Grid item key={product.id} xs={12} sm={6} lg={4}>
+        <ProductPreview key={product.id} product={product} />
+      </Grid>
+    );
+  })
+}
```

Comment:
Could this code be tested (we should add a unit test file as well to include all our function tests)?

Reply:
Same goes from `renderProductGrid`

Reply:
We can do some unit tests, sure! What would the condition be though? Just making sure that something was rendered to the screen? Like making sure the text is visible or something?

Reply:
Yup exactly. Take a lot at template tests online, it should be similar to what you'd be looking for here.

Reply:
I replaced `renderProductGrid` with the `ProductGrid` component for easier testing and best practices. I added the unit test to App.test.js, and it checks the screen to ensure all product names are visible. We can add different tests later but this helps us narrow down where a future issue may be coming from.